starter_code/src/movies.js

```diff
@@ -1,20 +1,110 @@
-/* eslint no-restricted-globals: 'off' */
 // Turn duration of the movies from hours to minutes 
 
 
+function turnHoursToMinutes(movies) {
+  
+  return movies.map(movie => {
+    let minsDuration = 0
+    movie.duration.split(' ').forEach((el) => {
+      if(el.includes('h')) minsDuration += parseInt(el) * 60
+      if(el.includes('min')) minsDuration += parseInt(el)
+    })
+
+    return {
+      title: movie.title,
+      year: movie.year,
+      director: movie.director,
+      duration: minsDuration,
+      genre: movie.genre,
+      rate: movie.rate
+    }
+  })
+}
+
 // Get the average of all rates with 2 decimals 
+function ratesAverage(movies) {
+
+  //return (movies.reduce((sum, movie) => sum + parseFloat(movie.rate), 0).toFixed(2) / movies.lenght)
 
+  let x = movies.reduce((suma, movie) => {
+    return suma + parseFloat(movie.rate)
```

Comment:
Aqui lo más seguro es usar `Number()` en lugar de `parseFloat()`. Si pasas una cadena vacía, `parseFloat()` regresa `NaN`, pero `Number("")` regresa `0`. Por eso hay un error en esta prueba de Jasmine: `Return Average even if one of the movies does not have rate!`